frmts/georaster/georaster_wrapper.cpp

```diff
@@ -539,6 +539,66 @@ GeoRasterWrapper *GeoRasterWrapper::Open(const char *pszStringId, bool bUpdate)
     return poGRW;
 }
 
+bool ValidateInsertExpression(const CPLString &sInsertStatement);
+bool ValidateDescriptionExpression(const CPLString &sInsertStatement);
+
+bool ValidateInsertExpression(const CPLString &sInsertStatement)
+{
+    std::vector<char> vStringExpressions;
+    for (size_t nPos = 0; nPos < sInsertStatement.length(); ++nPos)
+    {
+        // Check if text is inside quotes
+        if (sInsertStatement[nPos] == '\'' &&
+            sInsertStatement[nPos - 1] != '\\')
```

Comment:
Could nPos-1 < 0? 
Have you checked if the quote in the string is "'" or "\'"?

Reply:
As for the check for the type of quote.
This check accounts for the beginning and end of a quote, quotes inside quotes must be escaped with \ and these are not considered as an opening or as a closing quote. So, this considers the following cases:
'str;--//ing': This is okay, they are inside a quote
'str\';;': This is also okay, the inner quote does not count as a closing quote.
 

Take the following strings:
1. `(1243, '123', '\'hello;\'')`: This is valid because the `;` is stored inside an outer quote, the quote \' after the opening quote does not close the quote.
2. `(1243, '123', '\'hello\';')`: This is also valid because the `;` is stored inside an outer quote.
3. `(1243, '123', '\'hello\'');`: This is not valid because the `;` is outside a quote.

Reply:
in database ' is not escaped by \. So none of your strings with \' works. 
Try the following in the sqlplus to see what happens: 

select '\'hello;\'' from dual;
select '\'hello\';' from dual;
select '\'hello\'' from dual;

Reply:
You are right, my mistake. Going to fix this ASAP

Reply:
Commit 0992659 contains a fix for this. :)